control-plane/agents/core/src/volume/specs.rs

```diff
@@ -662,23 +671,25 @@ impl ResourceSpecsLocked {
         request: &SetVolumeReplica,
     ) -> Result<Volume, SvcError> {
         let spec = self
-            .get_volume(&request.uuid)
+            .get_locked_volume(&request.uuid)
             .context(errors::VolumeNotFound {
                 vol_id: request.uuid.to_string(),
             })?;
-        let status = registry.get_volume_status(&request.uuid).await?;
+        let state = registry.get_volume_state(&request.uuid).await?;
 
         let operation = VolumeOperation::SetReplica(request.replicas);
-        let spec_clone = SpecOperations::start_update(registry, &spec, &status, operation).await?;
+        let spec_clone = SpecOperations::start_update(registry, &spec, &state, operation).await?;
 
         assert_ne!(request.replicas, spec_clone.num_replicas);
         if request.replicas > spec_clone.num_replicas {
-            self.increase_volume_replica(registry, spec, status, spec_clone)
-                .await
+            self.increase_volume_replica(registry, spec, state, spec_clone.clone())
+                .await?
         } else {
-            self.decrease_volume_replica(registry, spec, status, spec_clone)
-                .await
-        }
+            self.decrease_volume_replica(registry, spec, state, spec_clone.clone())
+                .await?
+        };
+
+        registry.get_volume(&request.uuid).await
```

Comment:
don't the increase/decrease return the volume already?

Reply:
They didn't, but I've changed them now so that they do.